Sources/SwiftProtobuf/ProtobufBinaryTypes.swift

```diff
@@ -789,14 +777,21 @@ public extension ProtobufBinaryMessageBase {
 
 public extension ProtobufMessage {
     static func decodeProtobufMapValue(decoder: inout ProtobufFieldDecoder, value: inout Self?) throws {
-        let handled = try decoder.decodeSingularMessageField(fieldType: Self.self, value: &value)
-        assert(handled)
+        try decoder.decodeSingularMessageField(fieldType: Self.self, value: &value)
+        assert(value != nil)
     }
 
     init(protobuf: Data) throws {
         try self.init(protobuf: protobuf, extensions: nil)
     }
 
+    init(protobufBuffer: UnsafeBufferPointer<UInt8>) throws {
+        try self.init(protobufBuffer: protobufBuffer, extensions: nil)
+    }
+}
+
+// For Proto3 messages, we need to preserve the unknown fields when we're done
```

Comment:
> We already have the ProtobufExtensibleMessage protocol that extends ProtobufMessage, so I wonder if we can leverage that further by putting extension-set-specific initializers/methods there instead of creating more protocols.

I have some ideas along these lines that I want to explore after I land this change.